### PR TITLE
[WIP] Fixing an Infinite Loop case in UnmatchedChecker.

### PR DESCRIPTION
**Pull request by MarisaKirisame**
the below code will cause the unmatched case in type checker to hang. The problem is that the expand function will contract but it shouldnt. Will upload a fix tmr.
%P~A9741EKF$9@6LAV{B ZA
W E(V~9 E4DM`BU_U3@GBOE

### CODE REVIEW

**Comment by MarisaKirisame** [comment]
@slyubomirsky

**Comment by slyubomirsky** [comment]
Excellent catch, thank you for finding a repro.

As we privately discussed, this appears to be a problem in expanding the wildcards: there is a line that potentially erases information from candidates, violating the invariant that expanded candidates should always be larger than the original ones. This should essentially require a one-line change in ExpandWildcards (I will let you get to it)

**Comment by slyubomirsky** [comment]
@MarisaKirisame are you planning to update, or should I make the changes myself?

**Comment by MarisaKirisame** [comment]
I will do it myself. Was hoping mike will try it...

**Comment by slyubomirsky** [comment]
@MarisaKirisame I hate to be a nuisance, but I wanted to give another reminder so we don't forget to fix this case

**Comment by MarisaKirisame** [comment]
@slyubomirsky @zhiics @junrushao1994 can you guys review?

File: tests/python/relay/test_pass_unmatched_cases.py [hunk 1]
```diff
@@ -273,3 +274,27 @@ def test_tuple_match():
     clause = relay.Clause(relay.PatternTuple([relay.PatternVar(a), relay.PatternVar(b)]), a + b)
     x = relay.Match(relay.Tuple([relay.const(1), relay.const(1)]), [clause])
     assert len(unmatched_cases(x)) == 0
+
+
+def test_inf_loop_case():
+    code = """
```
Comment (zhiics) [inline tests/python/relay/test_pass_unmatched_cases.py]: Should this be aligned in the raw string? Not sure if it will cause parsing problem.

**Pull request merged**